### PR TITLE
v4.0.0 - Rewrite entire library

### PR DESCRIPTION
**Pull request by joshuasing**
This pull request entirely rewrites UpdateLib, to improve the code style and quality.
This pull request also does the following:
 - Move to `MIT License`
 - Marks < 4.0 as unsupported in SECURITY.md
 - Bumps the version to `4.0.0`
 - Uses the maven-gpg-plugin to sign artifacts.
 - Improves version comparison

Before this pull request is merged, the following should be completed:
 - [x] Update and move documentation.

### CODE REVIEW

File: src/main/java/dev/hypera/updatelib/comparators/IVersionComparator.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * SOFTWARE.
+ */
+
+package dev.hypera.updatelib.comparators;
+
+import dev.hypera.updatelib.exceptions.VersionComparisonFailureException;
+import dev.hypera.updatelib.objects.enums.Status;
+import org.jetbrains.annotations.NotNull;
+
+public interface IVersionComparator {
```
Comment (SLLCoding) [inline src/main/java/dev/hypera/updatelib/comparators/IVersionComparator.java:30]: Could this implement Comparator?

**Review by SLLCoding — APPROVED** [review]

**Comment by joshuasing** [comment]
Updated documentation: https://wiki.hypera.dev/updatelib/

File: src/main/java/dev/hypera/updatelib/comparators/impl/CalendarVersioningComparator.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	@Override
+	public @NotNull Status compareVersions(@NotNull String currentVersion, @NotNull String distributedVersion) throws VersionComparisonFailureException {
+		try {
+			return currentVersion.compareTo(distributedVersion) <= 0 ? Status.UNAVAILABLE : Status.AVAILABLE;
+		} catch (Exception ex) {
+			throw new VersionComparisonFailureException(ex);
+		}
+	}
+
+	public static void main(String[] args) {
```
Comment (SLLCoding) [inline src/main/java/dev/hypera/updatelib/comparators/impl/CalendarVersioningComparator.java:54]: Left in a debug method?

**Review by SLLCoding — APPROVED** [review]

File: src/main/java/dev/hypera/updatelib/comparators/impl/CalendarVersioningComparator.java (hunk 2 above)
Comment (joshuasing) [reply src/main/java/dev/hypera/updatelib/comparators/impl/CalendarVersioningComparator.java:54]: Yup, oops

**Review by SLLCoding — APPROVED** [review]

**Pull request merged**